packages/workflow/src/internals.ts

```diff
@@ -611,7 +617,13 @@ export class Activator implements ActivationHandler {
 
   // Intentionally non-async function so this handler doesn't show up in the stack trace
   protected queryWorkflowNextHandler({ queryName, args }: QueryInput): Promise<unknown> {
-    const fn = this.queryHandlers.get(queryName)?.handler;
+    let fn = this.queryHandlers.get(queryName)?.handler;
+    // Fallback to default query handler if no handler is registered.
+    if (fn === undefined && this.defaultQueryHandler !== undefined) {
+      // Use non-null assertion because Typescript can't tell that defaultQueryHandler cannot be undefined here.
+      fn = (...args) => this.defaultQueryHandler!(queryName, ...args);
```

Comment:
TS is technically right: even though you just checked that `this.defaultQueryHandler` is not undefined, you are then using that field inside a function call; there is no guarantee that `this.defaultQueryHandler` will still be valid by the time `fn` gets called. Well, we know it will, but the compiler can't figure this out.

You can avoid the TS error without a non-null assertion by copying `this.defaultQueryHandler` to a temporary variable inside the `if` block, or by using `this.defaultQueryHandler.bind(...)` instead:

```suggestion
      fn = this.defaultQueryHandler.bind(this, queryName);
```

Reply:
Done